### DISCUSSION TITLE
Adding a content bundle to a different project in the solution

### DISCUSSION THREAD

**Discussion by Wolftr** [MonoGame/MonoGame#8203] [Content]
I apologize for the somewhat bad title but it is difficult to explain what I am asking without going into more detail. I am attempting to make a game engine using MonoGame as a base. I would like to be able to drag and drop the .dll for my project as a reference into any other project I may want to use it for, but the Content.mgcb file only exists in the engine. I would like it to exist within the game only but I am not sure how to create a new one that can be loaded in upon starting the game. I tried copying it from one directory to the other but the editor will not open when I do that. Is there some sort of method or project setup I can use to allow me to manage and build content from my game project rather than my engine project?

**Comment by ThomasFOG** [comment]
I'm not sure if that answers exactly to your question, but for a start, .mgcb files can be detached from any runtime engine.

.mgcb files get compiled by adding a reference to ```MonoGame.Content.Builder.Task```. The reference to ```MonoGame.Framework``` is not required for .mgcb files to build (and the runtime engine doesn't need the reference to Builder.Task to work).

So at least you can have the two separated. However, .mgcb stuff is designed to be compiled at build time, not runtime.
You could however, build MGCB as a standalone tool that could be called by your engine so that it produces XNB without the need of a .mgcb file (or alternatively auto-generating it). But mind that it is not designed to be shipped with games, if the intent is to build content when a game launches (though it's fine to ship that with an engine or tools).

**Reply by Wolftr** [reply]
With a little more understanding of what mgcb files are actually doing I know how to say what I'm looking for a little more precisely.
I have two projects, one for the engine and one for the game. In the future I plan on having the engine project not be included in the solution, rather opting to use a DLL library. I want my content to be located in the "Test Game" project and to be able to edit said file by double clicking it in the solution explorer like I can for the other one. 

![image](https://github.com/MonoGame/MonoGame/assets/37937456/bb367490-322c-43d7-bee4-10826dc1e4c4)

I have two known issues with this setup. The first is I cannot edit the file when I double click it in Visual Studio. I assume I am missing some setting that would allow for that. This was my original issue that made me post the question.
The second is new after information I got from another answer and it is a lot less clear. After adding `MonoGame.Content.Builder.Task` to "Test Game" and adding to my project file
```
<ItemGroup>
  <MonoGameContentReference Include="path/to/file.mgcb" />
</ItemGroup>
```
I get a not very descriptive error
```
Error	MSB3073	The command "dotnet mgcb /quiet /@:"F:\Coding Projects\VulfEngine\TestGame\Content\Content.mgcb" /platform:DesktopGL /outputDir:"F:/Coding Projects/VulfEngine/TestGame/Content/bin/DesktopGL/Content" /intermediateDir:"F:/Coding Projects/VulfEngine/TestGame/Content/obj/DesktopGL/net6.0/Content" /workingDir:"F:/Coding Projects/VulfEngine/TestGame/Content/"" exited with code 1.
```
Clearly something is wrong with my setup that is causing the build to fail. I have both MonoGame packages my engine project has referenced (DesktopGL and Content.Builder) and when looking at both project files nothing seems to be missing from my "Test Game" file. Beyond that, I am not sure what could be causing the file to not build.

**Comment by harry-cpp** [comment]
Extra note, if you want to include a `.mgcb` files thats outside of the project folder, after referencing the `MonoGame.Content.Builder.Task` nuget, you can do so by manually adding the following to the project file:
```xml
<ItemGroup>
  <MonoGameContentReference Include="path/to/file.mgcb" />
</ItemGroup>
```
